src/services/mongoose/wallets.ts

```diff
@@ -29,10 +34,34 @@ export const WalletsRepository = (): IWalletsRepository => {
     const account = await AccountsRepository().findById(accountId)
     // verify that the account exist
     if (account instanceof Error) return account
-
     try {
+      // FLASH FORK: create IBEX account if currency is USD
+      let ibexAccountId = ""
+      if (currency === "USD") {
+        const IbexAccountCreationResponse = await requestIBexPlugin(
+          "POST",
+          IbexRoutes.API_CreateAccount,
+          {},
+          {
+            name: accountId,
+            currencyId: 3,
+          },
+        )
+        console.log("IbexAccountCreationResponse", IbexAccountCreationResponse)
+        if (
+          !IbexAccountCreationResponse ||
+          !IbexAccountCreationResponse.data ||
+          !IbexAccountCreationResponse.data["data"]["id"]
+        ) {
+          console.error({ error: "unable to get IbexAccountCreationResponse" })
```

Comment:
should we throw an error here if it fails cause then we will create a new wallet with an empty `ibexAccountId`

Reply:
what is the alternative? Doesn't this like take care of the empty ibexAccountId scenario?

 `id: currency === "USD" ? ibexAccountId : crypto.randomUUID(),`

Reply:
ya that should handle the scenario, I'm just not sure if we would want a random uuid but if that works for you then I'm good with it aswell

Reply:
wait that doesnt take care of the empty ibexAccountId scenario because it will use the empty one if `currency === "USD"` we would want `currency === "USD" && !!ibexAccountId`